x-pack/plugins/profiling/server/lib/setup/fleet_policies.ts

```diff
@@ -182,3 +184,35 @@ export async function createSymbolizerPackagePolicy({
     force: true,
   });
 }
+
+export async function validateProfilingInApmPackagePolicy({
+  soClient,
+  packagePolicyClient,
+}: ProfilingSetupOptions): Promise<PartialSetupState> {
+  const apmPolicy = await getApmPolicy({ packagePolicyClient, soClient });
+
+  return {
+    policies: {
+      apm: {
+        profilingEnabled: !!(
+          apmPolicy && apmPolicy?.inputs[0].config?.['apm-server'].value?.profiling
```

Comment:
The only concern I have here is potential changes to the policy object in the future. Is there a more type-safe way to express this in TypeScript? I don't think the policy object will change between now and soon after GA, but I just wanted to raise the concern. Feel free to deprioritize.

Reply:
Unfortunately not. In other plugins, we have the same issue/concern.